bin/generate-phpstorm-stubs-map.php

```diff
@@ -0,0 +1,144 @@
+{
+    /** @var string[] */
+    private $classNames = [];
+
+    /** @var string[] */
+    private $functionNames = [];
+
+    public function enterNode(Node $node) : ?int
+    {
+        if ($node instanceof Node\Stmt\ClassLike) {
+            $this->classNames[] = $node->namespacedName->toString();
+
+            return NodeTraverser::DONT_TRAVERSE_CHILDREN;
+        }
+
+        if ($node instanceof Node\Stmt\Function_) {
+            /** @psalm-suppress UndefinedPropertyFetch */
+            $this->functionNames[] = $node->namespacedName->toString();
+
+            return NodeTraverser::DONT_TRAVERSE_CHILDREN;
+        }
+
+        return null;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getClassNames() : array
+    {
+        return $this->classNames;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getFunctionNames() : array
+    {
+        return $this->functionNames;
+    }
+
+    public function clear() : void
+    {
+        $this->classNames    = [];
+        $this->functionNames = [];
+    }
+};
+
+$nodeTraverser = new NodeTraverser();
+$nodeTraverser->addVisitor(new NameResolver());
+$nodeTraverser->addVisitor($fileVisitor);
+
+$map = ['classes' => [], 'functions' => []];
+
+foreach (new DirectoryIterator($phpStormStubsDirectory) as $directoryInfo) {
+    if ($directoryInfo->isDot()) {
+        continue;
+    }
+
+    if (! $directoryInfo->isDir()) {
+        continue;
+    }
+
+    if (in_array($directoryInfo->getBasename(), ['tests', 'meta'], true)) {
+        continue;
+    }
+
+    foreach (new DirectoryIterator($directoryInfo->getPathName()) as $fileInfo) {
+        if ($fileInfo->isDot()) {
+            continue;
+        }
+
+        if (! preg_match('/\.php$/', $fileInfo->getBasename())) {
+            continue;
+        }
+
+        $ast = $phpParser->parse(file_get_contents($fileInfo->getPathname()));
+
+        $nodeTraverser->traverse($ast);
+
+        /** @psalm-suppress UndefinedMethod */
+        foreach ($fileVisitor->getClassNames() as $className) {
+            $map['classes'][$className] = $fileInfo->getPathname();
+        }
+
+        /** @psalm-suppress UndefinedMethod */
+        foreach ($fileVisitor->getFunctionNames() as $functionName) {
+            $map['functions'][$functionName] = $fileInfo->getPathname();
+        }
+
+        $fileVisitor->clear();
+    }
+}
+
+$uniqueFiles = [];
+foreach ($map as $typeFiles) {
+    foreach ($typeFiles as $file) {
+        $uniqueFiles[$file] = null;
+    }
+}
+
+ksort($uniqueFiles);
+
+$fileNumbers = array_flip(array_keys($uniqueFiles));
+
+$output = "<?php\n\ndeclare(strict_types=1);\n\n\$files = [\n";
+foreach ($uniqueFiles as $fileName => $tmp) {
+    $relativeFileName = str_replace('\\', '/', substr($fileName, strlen($phpStormStubsDirectory) + 1));
+
+    $output .= sprintf('    \'%s\',%s', $relativeFileName, "\n");
+}
+$output .= "];\n\n";
+
+$output .= "return [\n";
```

Comment:
This entire section should be a single `var_export()`

Reply:
The `var_export` export is ugly :) If I change it to static class with constants, it should then be compatible with CS so it's easier to genereted by myself.

Reply:
It may as well be ugly, but it works and it will prevent any escaping issues :P

Let's please not write anything around `\\` and stuff like that, and just use `var_export()`

Reply:
Done. And fixed formatting automatically via `phpcbf` :)